api-platform/core/3.1/config/packages/api_platform.yaml

```diff
@@ -0,0 +1,10 @@
+api_platform:
+    title: Hello API Platform
+    version: 1.0.0
+    # Good defaults for REST APIs
+    defaults:
+        stateless: true
```

Comment:
This might be too opinionated? If you create a frontend on the same domain that logs in via cookies, then authentication wouldn't work, correct?

Reply:
Cookies can be used, but the session cannot. It's one of [the mandatory constraints of the REST architecture](https://www.ics.uci.edu/%7Efielding/pubs/dissertation/rest_arch_style.htm#sec_5_1_3).

IMHO it's not opinionated, it's just encouraging the correct way of creating REST APIs (and web applications in general).

Better let the user remove this line only if they know what they are doing (and by doing that they are probably doing a mistake) than having them creating non-stateless "REST" APIs.

Reply:
By the way since 6.3 stateless routes automatically enable stateless firewalls, so enabling this feature shouldn't be an issue even when using the default config of firewalls: https://symfony.com/doc/current/reference/configuration/security.html#stateless

Reply:
but if this is the good thing to do, why isn't it the default in ApiPlatform then ?

Reply:
@stof it is: https://github.com/api-platform/api-platform/blob/main/api/config/packages/api_platform.yaml

Reply:
I'm talking about the default in the Configuration class of the bundle (i.e. the actual default value of the setting), not the config provided in the API platform distribution.

Reply:
Because this would be a BC break. We'll maybe do that for API Platform 4 next year.